substrate/client/chain-spec/src/lib.rs

```diff
@@ -16,19 +16,229 @@
+//!       <td>Optional map of <code>block_number</code> to <code>wasm_code</code> More details in
+//! material to follow.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>genesis</td>
+//!       <td>Defines the initial state of the runtime. More details in material to follow.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>code</td>
+//!       <td>The runtime wasm code blob in hex format.</td>
+//!     </tr>
+//!   </tbody>
+//! </table>
+//!
+//! # `genesis`: Initial Runtime State
+//!
+//! All nodes in the network must build subsequent blocks upon exactly the same genesis block.
+//!
+//! The information configured in the `genesis` section of a chain specification is used to build
+//! the genesis storage, which is essential for creating the genesis block, since the block header
+//! includes the storage root hash.
+//!
+//! The `genesis` key of the chain specification definition describes the
+//! initial state of the runtime. For example, it may contain:
+//! - an initial list of funded accounts,
+//! - the administrative account that controls the sudo key,
+//! - an initial authorities set for consensus, etc.
+//!
+//! The compiled WASM blob of the runtime code is stored in the state (and therefore on the chain).
+//! As a result, the initial runtime must also be provided within the chain specification (under the
+//! `code` field).
+//!
+//! The genesis state can be represented in the formats described by [`GenesisFormat<G>`]
+//! enum. In essence, the most important formats are:
+//! <table>
+//!   <thead>
+//!     <tr>
+//!       <th>Format</th>
+//!       <th>Description</th>
+//!     </tr>
+//!   </thead>
+//!   <tbody>
+//!     <tr>
+//!       <td>
+//! 		<a href="enum.GenesisFormat.html#variant.RuntimeGenesisConfig"><code>runtime</code></a>
+//!       </td>
+//!       <td>A JSON object that provides an explicit and comprehensive representation of the
+//! <code>RuntimeGenesisConfig</code> struct. Must contain all the keys of the genesis config, no
+//! defaults will be used.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>
+//! 		<a href="enum.GenesisFormat.html#variant.RuntimeGenesisConfigPatch"><code>patch</code></a>
+//!       </td>
+//!       <td>A JSON object that offers a partial representation of the
+//!       <code>RuntimeGenesisConfig</code> provided by the runtime. It contains a patch, which is
+//! essentially a list of keys to customize in the default runtime's
+//! <code>RuntimeGenesisConfig</code>.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>
+//! 		<a href="enum.GenesisFormat.html#variant.Raw"><code>raw</code></a>
+//!       </td>
+//!       <td>A JSON object with two fields: <code>top</code> and <code>children_default</code>.
+//! Each field is a map of <code>key => value</code> pairs representing entries in a genesis storage
+//! trie.</td>
+//!     </tr>
+//!   </tbody>
+//! </table>
+//!
+//! For production or long-lasting blockchains, using the `raw` format in the chain specification is
+//! recommended.
```

Comment:
Can you add why is recommended? I.e. current on-chain runtime may be not the same as genesis